Add docs best effort responses (#3865)
dfinity/portal: docs/developer-docs/defi/tokens/ledger/usage/icp_ledger_usage.mdx

        // the sender or receiver to attach additional information to the transaction.
        memo: memo.map(|m| Memo::from(m)),
        to,
        amount,
        // The ledger supports subaccounts. You can pick the subaccount of the caller canister's
        // account to use for transferring the ICP. If you don't specify a subaccount, the default
        // subaccount of the caller's account is used.
        from_subaccount,
        // The ICP ledger canister charges a fee for transfers, which is deducted from the
        // sender's account. The fee is fixed to 10_000 e8s (0.0001 ICP). We specify it here
        // to ensure that it hasn't changed. Using None would use whatever the current fee is, even
        // if it changes.
        fee: Some(NumTokens { 10_000 }),
        // The created_at_time is used for deduplication, which we don't use in this example as
        // we use unbounded-wait calls. You should, however, set it if you opt to use bounded-wait
        // calls, or if you use ingress messages, or if you are worried about bugs in the ICP
        // ledger. See the ICRC-1 example for how to do this.
        created_at_time: None,
    };

>     // We use an unbounded-wait call here as we assume that the ICP ledger is trusted and won't

Reviewer: ```suggestion
    // Use an unbounded-wait call here as we assume that the ICP ledger is trusted and won't
```

    // spin forever without producing a response. See the ICRC-1 usage doc if you want to use
    // bounded-wait calls
    match Call::unbounded_wait(icp_ledger, "icrc1_transfer")
        .with_arg(&args)
        .await
    {
        // The transfer call succeeded
        Ok(res) => match res.candid::<Result<BlockIndex, TransferError>>() {
            Ok(_i) => Ok(()),

>             // The ledger canister returned an error, for example because our balance was too low.

Reviewer: ```suggestion
            // The ledger canister returned an error, for example if the balance was too low.
```

>             // The transfer didn't happen, and we report an error back to the user.

Reviewer: ```suggestion
            // The transfer didn't happen, and it reports an error back to the user.
```

            // Look up the TransferError type in icrc_ledger_types for more details.
            Ok(Err(e)) => Err(format!("Ledger returned an error: {:?}", e)),
        }

>         // An unclean reject signals that something went wrong with our call, but the system isn't
>         // sure whether the call was executed. Since we issued an unbounded-wait call, this
>         // happens either because the ledger explicitly rejected our call, or because it panicked
>         // while processing our request.
>         // The ICP ledger doesn't explicitly reject calls. When using the ICRC-1 interface, it's also
>         // not intended to panic; instead it should report any errors at the "user-level", encoded
>         // in Candid. Here, we assume that it correctly implements this behavior and does't panic.
>         // However, if you don't want to make that assumption, you can add your own error handling
>         // of that case here.

Reviewer: ```suggestion
        // An unclean reject signals that something went wrong with the call, but the system isn't
        // sure whether the call was executed. Since it issued an unbounded-wait call, this
        // happens either because the ledger explicitly rejected the call, or because it panicked
        // while processing the request.
        // The ICP ledger doesn't explicitly reject calls. When using the ICRC-1 interface, it's also
        // not intended to panic; instead it should report any errors at the "user-level", encoded
        // in Candid. Here, it is assumed that it correctly implements this behavior and does't panic.
        // However, if you don't want to make that assumption, you can add your own error handling
        // of that case here.
```
